Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/InteropLibraryCreation.kt

```diff
@@ -44,12 +48,44 @@ fun createInteropLibrary(arguments: LibraryCreationArguments) {
             version,
             arguments.target
     ).apply {
-        // TODO: Add write strategy that splits big fragments.
-        val metadata = arguments.metadata.write()
+        val metadata = arguments.metadata.write(ChunkingWriteStrategy())
         addMetadata(SerializedMetadata(metadata.header, metadata.fragments, metadata.fragmentNames))
         addNativeBitcode(arguments.nativeBitcodePath)
         addManifestAddend(arguments.manifest)
         addLinkDependencies(arguments.dependencies)
         commit()
     }
 }
+
+// TODO: Consider adding it to kotlinx-metadata-klib.
+class ChunkingWriteStrategy(
+        private val classesChunkSize: Int = 128,
+        private val packagesChunkSize: Int = 128
+) : KlibModuleFragmentWriteStrategy {
+
+    override fun processPackageParts(parts: List<KmModuleFragment>): List<KmModuleFragment> {
+        val fqName = parts.firstOrNull { it.fqName != null }?.fqName
```

Comment:
Maybe ensure that all `fqName`s are the same?
Btw, why `fqName` can be`nul`? Is it guaranteed that there is at least one non-null `fqName` among `parts`?

Reply:
It is guaranteed that all these `fqName` are the same. I'll note it in `processPackageParts` documentation.
> Btw, why fqName can be `null`?
Actually, it [shouldn't](https://github.com/JetBrains/kotlin/blob/master/libraries/kotlinx-metadata/klib/src/kotlinx/metadata/klib/KlibModuleMetadata.kt#L112). Will add an assertion.